src/Parser/ElementalShaman/Modules/Talents/Ascendance.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class Ascendance extends Module {
+  numLavaBurstsCast = 0;
+  numLightningBoltsCast = 0;
+  numOtherCasts = 0;
+
+  numCasts = {
+    [SPELLS.LAVA_BURST.id]: 1,
+    [SPELLS.EARTH_SHOCK.id]: 1,
+    [SPELLS.ELEMENTAL_BLAST.id]: 1,
+    [SPELLS.FLAME_SHOCK.id]: 1,
+    [SPELLS.LIGHTNING_BOLT.id]: 1,
+    'others': 1,
+  };
+
+  on_initialized() {
+    this.active = this.owner.selectedCombatant.hasTalent(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) || this.owner.selectedCombatant.hasHands(ITEMS.SMOLDERING_HEART.id);
+  }
+
+  get rawUpdate() {
+    return this.owner.selectedCombatant.getBuffUptime(SPELLS.ASCENDANCE.id) / this.owner.fightDuration;
+  }
+
+  get AscendanceUptime() {
+    return this.rawUpdate;
+  }
+
+  on_byPlayer_cast(event) {
+    if (this.owner.selectedCombatant.hasBuff(SPELLS.ASCENDANCE.id, event.timestamp)) {
+      const spellId = event.ability.guid;
+      if (this.numCasts[spellId]) {
+        this.numCasts[spellId]++;
+      } else {
+        this.numCasts.others++;
+      }
+    }
+  }
+
+  on_finished() {
+    for(const i in this.numCasts) {
+      this.numCasts[i]--;
```

Comment:
Why are you doing this instead of starting the casts at 0?

Reply:
Because I was lazy with the spell count check [(Line 41-45)](https://github.com/MartijnHols/WoWAnalyzer/pull/206/files/8c2a167e7cb9d0fe32b11d7b08d5529b9d901579#diff-dd2dd7ffdeff121650959ffd4d1b3ca6R41); The check for the specific spells [(Line 18-22)](https://github.com/MartijnHols/WoWAnalyzer/pull/206/files/8c2a167e7cb9d0fe32b11d7b08d5529b9d901579#diff-dd2dd7ffdeff121650959ffd4d1b3ca6R18) wouldnt not work that way (0 would be always false and only `other` would be increased without a rewrite).

Reply:
Line 41 can just check the value with a `!== undefined`, that would be a more obvious solution.